product_variant_configurator/models/product_configurator.py

```diff
@@ -0,0 +1,254 @@
+# -*- coding: utf-8 -*-
+# © 2015 Oihane Crucelaegui - AvanzOSC
+# © 2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# © 2016 ACSONE SA/NV
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3
+
+from openerp import api, fields, models, exceptions, _
+from openerp.addons import decimal_precision as dp
+
+
+class ProductConfigurator(models.AbstractModel):
+    _name = 'product.configurator'
+
+    product_tmpl_id = fields.Many2one(
+        string='Product Template',
+        comodel_name='product.template',
+        auto_join=True)
+    product_attribute_ids = fields.One2many(
+        comodel_name='product.configurator.attribute',
+        domain=lambda self: [("owner_model", "=", self._name)],
+        inverse_name='owner_id',
+        string='Product attributes',
+        copy=True)
+    price_extra = fields.Float(
+        compute='_compute_price_extra',
+        digits=dp.get_precision('Product Price'),
+        help="Price Extra: Extra price for the variant with the currently "
+             "selected attributes values on sale price. eg. 200 price extra, "
+             "1000 + 200 = 1200.")
+    product_id = fields.Many2one(
+        string="Product",
+        comodel_name="product.product")
+    name = fields.Char()
+
+    @api.depends('product_attribute_ids', 'product_attribute_ids.value_id')
+    def _compute_price_extra(self):
+        for record in self:
+            record.price_extra = sum(
+                record.mapped('product_attribute_ids.price_extra'))
+
+    @api.onchange('product_tmpl_id')
+    def onchange_product_tmpl_id(self):
+        if not self.product_tmpl_id:
+            return {}
+
+        # First, empty current list
+        self.product_attribute_ids = [
+            (2, x.id) for x in self.product_attribute_ids]
+        if not self.product_tmpl_id.attribute_line_ids:
+            self.product_id = \
+                self.product_tmpl_id.product_variant_ids[0].id
+        else:
+            if not self.env.context.get('not_reset_product'):
```

Comment:
I could not find where not_reset_product is set. @zakiuu @pedrobaeza do you know what it's for?